dask/array/tiledb_io.py

```diff
@@ -124,7 +132,7 @@ def to_tiledb(
 
     tiledb_config = storage_options or dict()
     # encryption key, if any
-    key = tiledb_config.pop("key", None)
+    key = tiledb_config.pop("key", key)
```

Comment:
Sorry, after taking another look through this morning, I think this logic should actually be inverted. Something like this:

```suggestion
    key = key or tiledb_config.pop("key", None)
```